### ISSUE TITLE
False positive: SC1035 You are missing a required space after the `!`

### ISSUE THREAD

**Issue by Bilge** [koalaman/shellcheck#2869]
#### For bugs
- SC1035
- online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC1035)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

user:is_superuser() ((!UID))
```

#### Here's what shellcheck currently says:
```
^-- [SC1035](https://www.shellcheck.net/wiki/SC1035) (error): You are missing a required space after the !.
```

#### Here's what I wanted or expected to see:
No error.

**Comment by koalaman** [comment]
This is unfortunately a known problem. You can work around it with `user:is_superuser() { ((!UID)); }`.

ShellCheck has the annoying restriction that all functions bodies must be `{ .. }` or `( .. )`, which is actually more work than the correct approach of allowing any compound command. I've tried to fix it many times over the years, but it keeps leading to strange and misleading error messages for a variety of newbie syntax issues as ShellCheck keeps going and talking about function definitions when the user really just botched an array assignment that happened to be followed by a for loop. Hopefully one day...
